### DISCUSSION TITLE
`record-progress` event for Record plugin

### DISCUSSION THREAD

**Discussion by aantusahaa** [katspaugh/wavesurfer.js#3431] [Ideas]
Hi,
Foremost, I want to thank you for this awesome library! I primarily use `wavesurfer.js` for Visualization and the `Record` plugin.
In the recording portion, I want to show the current progress of the recording in the UI. But right now there is no straight forward way of getting the time from the plugin. I have found this similar [discussion](https://github.com/katspaugh/wavesurfer.js/discussions/3378). But it doesn't really help me because my case also involves pause and resume of recording.

If we have an event namely `record-progress` similar to the `audioprocess` event of wavesurfer itself, we will be able to get that information easily. On a related note, having methods like `getCurrentTime()`, `getDuration()` might also be helpful.

**Comment by katspaugh** [comment]
Those are all good points! I'll happily review a PR and help in case of any questions.

**Reply by aantusahaa** [reply]
I will try to give it a shot!
